src/main/java/com/stripe/model/Invoice.java

```diff
@@ -1,503 +1,894 @@
+   * proration will take place. To ensure that the actual proration is calculated exactly the same
+   * as the previewed proration, you should pass a <code>proration_date</code> parameter when doing
+   * the actual subscription update. The value passed in should be the same as the <code>
+   * subscription_proration_date</code> returned on the upcoming invoice resource. The recommended
+   * way to get only the prorations being previewed is to consider only proration line items where
+   * <code>period[start]</code> is equal to the <code>subscription_proration_date</code> on the
+   * upcoming invoice resource.
    */
-  public Invoice delete(RequestOptions options) throws StripeException {
-    return request(RequestMethod.DELETE, instanceUrl(Invoice.class, this.id), null,
-        Invoice.class, options);
+  public static Invoice upcoming(Map<String, Object> params) throws StripeException {
+    return upcoming(params, (RequestOptions) null);
   }
-  // </editor-fold>
 
-  // <editor-fold desc="list">
   /**
-   * List all invoices.
+   * At any time, you can preview the upcoming invoice for a customer. This will show you all the
+   * charges that are pending, including subscription renewal charges, invoice item charges, etc. It
+   * will also show you any discount that is applicable to the customer.
+   *
+   * <p>Note that when you are viewing an upcoming invoice, you are simply viewing a preview – the
+   * invoice has not yet been created. As such, the upcoming invoice will not show up in invoice
+   * listing calls, and you cannot use the API to pay or edit the invoice. If you want to change the
+   * amount that your customer will be billed, you can add, remove, or update pending invoice items,
+   * or update the customer’s discount.
+   *
+   * <p>You can preview the effects of updating a subscription, including a preview of what
+   * proration will take place. To ensure that the actual proration is calculated exactly the same
+   * as the previewed proration, you should pass a <code>proration_date</code> parameter when doing
+   * the actual subscription update. The value passed in should be the same as the <code>
+   * subscription_proration_date</code> returned on the upcoming invoice resource. The recommended
+   * way to get only the prorations being previewed is to consider only proration line items where
+   * <code>period[start]</code> is equal to the <code>subscription_proration_date</code> on the
+   * upcoming invoice resource.
    */
-  public static InvoiceCollection list(Map<String, Object> params) throws StripeException {
-    return list(params, null);
+  public static Invoice upcoming(Map<String, Object> params, RequestOptions options)
+      throws StripeException {
+    String url = String.format("%s%s", Stripe.getApiBase(), "/v1/invoices/upcoming");
+    return request(ApiResource.RequestMethod.GET, url, params, Invoice.class, options);
```

Comment:
7) Retrieve (custom): Static method retrieving a resource of a particular attribute

Reply:
https://github.com/stripe/stripe-java/blob/2aa5db5c39dc6e91228507798083f373904cc8c0/src/main/java/com/stripe/model/Invoice.java#L453-L477